src/redux/stores/collection/collectionApi.ts

```diff
@@ -1,54 +1,46 @@
+import { getTokenFromContract } from '@airswap/metadata';
+import { TokenInfo } from '@airswap/typescript';
 import { Web3Provider } from '@ethersproject/providers';
-import ERC721 from '@openzeppelin/contracts/build/contracts/ERC721.json';
 import { createAsyncThunk } from '@reduxjs/toolkit';
-import { Contract, ethers } from 'ethers';
 
 import { CollectionToken } from '../../../entities/CollectionToken/CollectionToken';
-import { transformErc721TokenToCollectionToken } from '../../../entities/CollectionToken/CollectionTokenTransformers';
-import { Erc721Token } from '../../../entities/Erc721Token/Erc721Token';
-
-export const getCollectionErc721Contract = (library: Web3Provider, token: string): Contract => {
-  const erc721Interface = new ethers.utils.Interface(ERC721.abi);
-
-  return new Contract(token, erc721Interface, library);
-};
+import { transformNFTTokenToCollectionToken } from '../../../entities/CollectionToken/CollectionTokenTransformers';
 
 interface fetchNFTMetadataParams {
   library: Web3Provider,
   collectionToken: string,
   startIndex: number,
 }
 
-export const fetchNFTMetadata = createAsyncThunk<
-CollectionToken[], fetchNFTMetadataParams>(
+export const fetchCollectionTokens = createAsyncThunk<(
+CollectionToken | undefined)[], fetchNFTMetadataParams>(
   'collection/fetchNFTMetadata',
   async ({ library, collectionToken, startIndex }) => {
-    // TODO: Add support for ERC-1155
-    const collectionContract = getCollectionErc721Contract(library, collectionToken);
-
-    if (!collectionContract) {
-      throw new Error('No collection contract found');
-    }
-
     const CHUNK_SIZE = 20;
     const tokensToFetch = new Array(CHUNK_SIZE)
       .fill(null)
       .map((value, index) => startIndex + index);
 
     const dataPromises = tokensToFetch.map(async (tokenId) => {
-      const tokenURI = await collectionContract.tokenURI(tokenId);
-      const res = await fetch(
-        tokenURI.replace('ipfs://', 'https://ipfs.io/ipfs/'),
-      );
-      const token = await res.json() as Erc721Token;
-
-      return transformErc721TokenToCollectionToken(
-        token,
-        tokenId,
-        0.154,
-      );
+      let tokenInfo: TokenInfo;
+
+      try {
+        tokenInfo = await getTokenFromContract(library, collectionToken, tokenId.toString());
+      } catch (e) {
+        console.error(new Error(`Unable to fetch data for ${collectionToken} with id ${tokenId}`));
+
+        return undefined;
+      }
+
+      const token = transformNFTTokenToCollectionToken(tokenInfo, tokenId, 0.154);
```

Comment:
The price will eventually wind up being a string from an indexer, might as well sort that now.

`'0154541201556702705'` is what I've been using and our guy @gpxl-dev made a nice npm package for dealing with them `@greypixel_/nicenumbers`

Reply:
Let's do the indexer stuff in a separate ticket